solver/src/solver.rs

```diff
@@ -47,7 +48,11 @@ pub trait Solver: 'static {
     /// order) so that they can be merged by the driver at its leisure.
     ///
     /// id identifies this instance of solving by the driver in which it invokes all solvers.
-    async fn solve(&self, auction: Auction) -> Result<Vec<Settlement>>;
+    async fn solve(
+        &self,
+        auction: Auction,
+        metrics: Arc<dyn SolverMetrics>,
```

Comment:
Yes!

Reply:
I also don't think it fits. If we want to instrument solvers with timing metrics, we can always create a wrapping solver implementation for that:
```rust
pub struct InstrumentedSolver {
    inner: Arc<dyn Solver>,
    metrics: Arc<dyn SolverMetrics>,
}

impl Solver for InstrumentedSolver {
    fn solve(...) -> ... {
        let result = self.inner.solve(...);
        self.metrics.do_stuff();
        result
    }
}
```

I feel like otherwise, we would end up having a lot of duplicated code for instrumenting the "common bits" for all the solvers (i.e. recording metrics common to all solvers).

For this PR though, the metrics are very specific to `SingleOrderSolver`s.